src/tk/wurst_client/mods/BaseFinderMod.java

```diff
@@ -60,94 +69,51 @@ public void onUpdate()
 		if(hasTimePassedM(3000))
 		{
 			matchingBlocks.clear();
+			
+			int playerX = (int) Math.floor(mc.thePlayer.posX);
+			int playerY = (int) Math.floor(mc.thePlayer.posY);
+			int playerZ = (int) Math.floor(mc.thePlayer.posZ);
+
 			for(int y = range; y >= -range; y--)
-			{
 				for(int x = range; x >= -range; x--)
-				{
 					for(int z = range; z >= -range; z--)
 					{
-						int posX = (int)(mc.thePlayer.posX + x);
-						int posY = (int)(mc.thePlayer.posY + y);
-						int posZ = (int)(mc.thePlayer.posZ + z);
+						int posX = playerX + x;
```

Comment:
As the variables aren't used outside of the loop there is technically no way they could be used without having a value assigned to them. There is a difference between a variable that is created and destroyed 1024 times (total: **2048** times) and a variable that's been created and destroyed 1 time and that has been assigned 1024 times (total: **1026** times). Even though Java is so ridiculously slow and you won't notice a difference in execution time in I'd call it `good coding style`.

But if you want to you can leave it like that as the compiler will move it out of the loop any ways (at least it should). Keeping it like that is much easier to maintain.